src/main/java/net/fabricmc/loader/impl/launch/knot/KnotClassDelegate.java

```diff
@@ -222,9 +222,13 @@ Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
 						URL url = parentClassLoader.getResource(fileName);
 
 						if (url == null) { // no .class file
-							String msg = "can't find class "+name;
-							if (LOG_CLASS_LOAD_ERRORS) Log.warn(LogCategory.KNOT, msg);
-							throw new ClassNotFoundException(msg);
+							try {
+								c = PLATFORM_CLASS_LOADER.loadClass(name);
```

Comment:
Can the parent class loader define resource-less classes? If that's the case, I recommend merging this branch with the `else` after `(!isValidParentUrl(url, fileName))`, like `if (url != null && !isValidParentUrl(url, fileName)) {} else {}`

Reply:
I don't think so, these seem to all come from the bootstrap CL.